core/src/main/java/com/google/instrumentation/trace/SpanId.java

```diff
@@ -53,35 +100,31 @@ public boolean equals(Object obj) {
     }
 
     SpanId that = (SpanId) obj;
-    return spanId == that.spanId;
+    return Arrays.equals(bytes, that.bytes);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hashCode(spanId);
+    return Arrays.hashCode(bytes);
   }
 
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
-        .add("spanId", new Formatter(Locale.US).format("%016x", spanId))
+        .add(
+            "spanId",
+            BaseEncoding.base16().lowerCase().encode(bytes))
         .toString();
   }
-  /**
-   * Returns whether the span identifier is valid. A valid span identifer is non-zero long value.
-   *
-   * @return whether the span identifier is valid.
-   */
-  public boolean isValid() {
-    return spanId != INVALID_SPAN_ID;
-  }
 
-  /**
-   * Returns the long that represents the span identifier.
-   *
-   * @return the span identifier.
-   */
-  public long getSpanId() {
-    return spanId;
+  @Override
+  public int compareTo(SpanId that) {
+    for (int i = 0; i < SPAN_ID_SIZE; i++) {
+      if (bytes[i] == that.bytes[i]) {
```

Comment:
You could save a line here:

if (bytes[i] != that.bytes[i]) {
  return bytes[i] < that.bytes[i] ? -1 : 1;
}